homeassistant/components/zwave_js/discovery.py

```diff
@@ -1342,10 +1359,9 @@ def async_discover_single_value(
             entity_category=schema.entity_category,
         )
 
+        # prevent re-discovery of the (primary) value if not allowed
         if not schema.allow_multi:
-            # return early since this value may not be discovered
-            # by other schemas/platforms
-            return
+            discovered_value_ids[device.id].add(value.value_id)
```

Comment:
We need to add the value id to the set of discovered value ids when the for loop is finished regardless of what happend earlier. Otherwise we'll potentially discover the value again.